feature-libs/organization/user-registration/components/form/user-registration-form.service.ts

```diff
@@ -0,0 +1,193 @@
+    return this.form.get('region.isocode') as FormControl;
+  }
+
+  /**
+   * Gets all title codes.
+   */
+  getTitles(): Observable<Title[]> {
+    return this.userRegisterFacade.getTitles();
+  }
+
+  /**
+   * Gets all countries list.
+   */
+  getCountries(): Observable<Country[]> {
+    return this.userAddressService.getDeliveryCountries().pipe(
+      tap((countries: Country[]) => {
+        if (Object.keys(countries).length === 0) {
+          this.userAddressService.loadDeliveryCountries();
+        }
+      })
+    );
+  }
+
+  /**
+   * Gets all regions list for specific selected country.
+   */
+  getRegions(): Observable<Region[]> {
+    return this.countryControl.valueChanges.pipe(
+      filter((countryIsoCode) => Boolean(countryIsoCode)),
+      switchMap((countryIsoCode) => {
+        this.regionControl.reset();
+        return this.userAddressService.getRegions(countryIsoCode);
+      })
+    );
+  }
+
+  /**
+   * Takes form values and builds custom message content.
+   */
+  protected buildMessageContent(form: FormGroup): Observable<string> {
+    return this.translationService.translate(
+      'userRegistrationForm.messageToApproverTemplate',
+      {
+        phoneNumber: form.get('phoneNumber')?.value,
+        addresLine: form.get('line1')?.value,
+        secondAddressLine: form.get('line2')?.value,
+        city: form.get('city')?.value,
+        state: form.get('region')?.get('isocode')?.value,
+        postalCode: form.get('postalCode')?.value,
+        country: form.get('country')?.get('isocode')?.value,
+        message: form.get('message')?.value,
+      }
+    );
+  }
+
+  /**
+   * Displays confirmation global message.
+   */
+  protected displayGlobalMessage(): void {
+    return this.globalMessageService.add(
+      { key: 'userRegistrationForm.successFormSubmitMessage' },
+      GlobalMessageType.MSG_TYPE_CONFIRMATION
+    );
+  }
+
+  /**
+   * Redirects the user back to the login page.
+   *
+   * This only happens in case of the `ResourceOwnerPasswordFlow` OAuth flow.
+   */
+  protected redirectToLogin() {
+    if (
+      this.authConfigService.getOAuthFlow() ===
+      OAuthFlow.ResourceOwnerPasswordFlow
+    ) {
+      this.routingService.go({ cxRoute: 'login' });
+    }
+  }
+
+  /**
+   * Registers new organization user.
+   */
+  registerUser(form: FormGroup): Observable<OrganizationUserRegistration> {
+    return this.buildMessageContent(form).pipe(
+      take(1),
+      switchMap((message: string) =>
+        this.organizationUserRegistrationFacade.registerUser({
+          titleCode: form.get('titleCode')?.value,
+          firstName: form.get('firstName')?.value,
+          lastName: form.get('lastName')?.value,
+          email: form.get('email')?.value,
+          /*
+           * Removing all extra spaces from a string
+           * in case potential user does not provide
+           * all address details.
+           */
+          message: message.replace(/\s+/g, ' '),
```

Comment:
I'm not sure about converting it in this layer. IMO we should pass form data as it is, and then use `SERIALIZER` to align it to API-specify.